docs/how-to/namespace-transition.md

```diff
@@ -0,0 +1,286 @@
+[Recommended Migration](#recommended-migration) to ensure continued good
+functionality of your installed events system.
+
+**Note that the current instructions only cover installation based on the
+release from the `knative-gcp` repository. We will update the page with specific
+`gcloud` instructions as the release becomes available there as well.**
+
+## Recommended Migration
+
+1. Run the following commands before installing the new `v0.20` release. We
+   explain the purpose of each command in comments.
+
+   ```shell script
+   # create the new namespace by copying the information from the old one
+   kubectl get namespace cloud-run-events -o yaml | \
+     sed 's/  name: cloud-run-events/  name: events-system/g' | \
+     kubectl create -f -
+   # setup identical authentication by copying the existing secrets from the old namespace
+   kubectl get secrets -n cloud-run-events -o yaml | \
+     sed 's/  namespace: cloud-run-events/  namespace: events-system/g' | \
+     kubectl create -f -
+   # copy the existing config maps to maintain existing configurations
+   kubectl get configmaps -n cloud-run-events --field-selector metadata.name!=kube-root-ca.crt -o yaml | \
+     sed 's/  namespace: cloud-run-events/  namespace: events-system/g' | \
+     kubectl create -f -
+
+   # remove the old webhook and controller to prevent them from interfering with the new ones
+   kubectl delete deployment webhook -n cloud-run-events
+   kubectl delete service webhook -n cloud-run-events
+   kubectl delete deployment controller -n cloud-run-events
+   kubectl delete service controller -n cloud-run-events
+   ```
+
+   For your convenience, we also packaged these instructions as a shell script
+   and as a yaml file that you can execute as shown below.
+
+   ```shell script
+   # execute the setup commands using the provided script
+   bash namespace-transition.sh
+   ```
+
+   Or, if you prefer executing these commands as a job, apply the following yaml
+   file.
+
+   ```shell script
+   # execute the setup commands as a job
+   kubectl apply -f namespace-transition.yaml
+   ```
+
+   Once you confirm that the job has completed successfully, you can delete it.
+
+   ```shell script
+   # confirm that the job has completed successfully
+   kubectl get all -n cloud-run-events
+   ```
+
+   ```shell script
+   NAME                                              READY   STATUS        RESTARTS   AGE
+   pod/namespace-transition-xwrxk                    0/1     Completed     0          3m3s
+
+   NAME                                              COMPLETIONS   DURATION   AGE
+   job.batch/namespace-transition                    1/1           37s        3m3s
+   ```
+
+   ```shell script
+   # delete the job
+   kubectl delete -f namespace-transition.yaml
+   ```
+
+1. Upgrade to the new namespace by applying the changes from the `master` branch
+   or any of the releases from `v0.20` onward. This instructions are identical
```

Comment:
knative supports upgrading by a single minor version number, so these instructions, imo, should be very specific to indicate migration from 0.19 to 0.20. They should also probably not recommend using the master branch. https://knative.dev/docs/install/upgrade-installation/

Reply:
Done.